code/modules/power/lighting.dm

```diff
@@ -225,49 +234,59 @@
 	return ..()
 
 /obj/machinery/light/update_icon()
-
-	switch(status)		// set icon_states
-		if(LIGHT_OK)
-			icon_state = "[base_state][on]"
-		if(LIGHT_EMPTY)
-			icon_state = "[base_state]-empty"
-			on = FALSE
-		if(LIGHT_BURNED)
-			icon_state = "[base_state]-burned"
-			on = FALSE
-		if(LIGHT_BROKEN)
-			icon_state = "[base_state]-broken"
-			on = FALSE
-	return
+	if(special == 1)
```

Comment:
Special conditions, like this, is called 'Hacky Code' and generally isn't allowed anywhere and isn't good practice. Overload this proc with your emergency lights instead.